deps/rmutil/rm_assert.h

```diff
@@ -1,35 +1,48 @@
-/*
- * Copyright Redis Ltd. 2016 - present
- * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
- * the Server Side Public License v1 (SSPLv1).
- */
-
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
 #ifndef __REDISEARCH_ASSERT__
 #define __REDISEARCH_ASSERT__
 
 #include "redismodule.h"
 #include "module.h"
 
+// Not to be called directly, used by the macros below
+#define _RS_LOG_ASSERT_FMT(condition, fmt, ...)                                      \
+    if (__builtin_expect(!(condition), 0)) {                                         \
+        RedisModule_Log(RSDummyContext, "warning", (fmt), __VA_ARGS__);              \
+        RedisModule_Assert(condition); /* Crashes server and create a crash report*/ \
+    }
+
 #ifdef NDEBUG
 
-#define RS_LOG_ASSERT(ctx, condition, fmt, ...)    (__ASSERT_VOID_CAST (0))
-#define RS_LOG_ASSERT_STR(ctx, condition, str)     (__ASSERT_VOID_CAST (0))
+// NOP
+#define RS_LOG_ASSERT_FMT(condition, fmt, ...)
+#define RS_LOG_ASSERT(condition, str)
+#define RS_ASSERT(condition)
+#define RS_ABORT(str)
 
 #else
 
-#define RS_LOG_ASSERT_FMT(condition, fmt, ...)                                          \
-    if (__builtin_expect(!(condition), 0)) {                                            \
-        RedisModule_Log(RSDummyContext, "warning", (fmt), __VA_ARGS__);                 \
-        RedisModule_Assert(condition); /* Crashes server and create a crash report*/    \
-    } 
-
+#define RS_LOG_ASSERT_FMT(condition, fmt, ...) _RS_LOG_ASSERT_FMT(condition, fmt, __VA_ARGS__)
```

Comment:
It seems like the code can be:
```suggestion
#ifdef NDEBUG
#define RS_LOG_DEBUG_ASSERT_FMT(condition, fmt, ...)
#else
#define RS_LOG_DEBUG_ASSERT_FMT(condition, fmt, ...) _RS_LOG_ASSERT_FMT(condition, fmt, __VA_ARGS__)
#endif

#define RS_LOG_ASSERT(condition, str)  RS_LOG_DEBUG_ASSERT_FMT(condition, str "%s", "")
#define RS_ASSERT(condition) RS_LOG_DEBUG_ASSERT_FMT(condition, "Assertion failed: %s", #condition)
#define RS_ABORT(str) RS_LOG_DEBUG_ASSERT_FMT(0, "Aborting: %s", str)
```